### DISCUSSION TITLE
Auto return last expression in function

### DISCUSSION THREAD

**Discussion by iacore** [vlang/v#18603] [Ideas]
Since we already have `... or { 1 }`, maybe it's a good idea to have this as well, for consistency:

```
fn foo() int {
    1
}
```

This is especially useful in closure

```
f := fn () int { 1 }
println(f())

pred := fn (a int, b int) { a < b }
```

**Comment by JalonSolov** [comment]
This would only save typing the word `return`.  Are there any other benefits?

**Comment by iacore** [comment]
Consistency with other language constructs (e.g. `match` branch).

**Comment by eptx** [comment]
Is this request only for functions...or blocks generally...as with Rust? 
See https://doc.rust-lang.org/rust-by-example/expression.html
In V becomes...
```
y := {
        x_squared := x * x
        x_cube := x_squared * x
        // This expression will be assigned to y
        x_cube + x_squared + x
    }
```

**Comment by JalonSolov** [comment]
Then the semantics are different, as well.

Personally, I don't care for just having the last thing calculated as the value assigned to the variable.  It takes more "mental parsing" to make sure of what's going on.  Seems like an extra way to cause yourself more problems than it might solve.

**Comment by eptx** [comment]
> Then the semantics are different, as well.
> 
> Personally, I don't care for just having the last thing calculated as the value assigned to the variable. It takes more "mental parsing" to make sure of what's going on. Seems like an extra way to cause yourself more problems than it might solve.

I can see you your point . Is there a way to encapsulate a computed value for Y in the example code above, in a block...or what's the idiomatic V way.

**Comment by JalonSolov** [comment]
The simplest way would be to do what every other language does... make a function, and call it.
```
y := calc(x)

fn calc(x int) int {
    x_squared := x * x
    x_cube := x_squared * x
    return x_cube + x_squared + x
}
```
